packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -351,13 +384,99 @@ contract VotingV2 is
+        }
+        // todo: consider if we want to check if the most recent price request has been settled?
+
+        spamDeletionProposals.push(SpamDeletionRequest(spamRequestIndices, currentTime, false, msg.sender));
+        uint256 proposalId = spamDeletionProposals.length - 1;
+
+        bytes32 identifier = SpamGuardIdentifierLib._constructIdentifier(proposalId);
+
+        _requestPrice(identifier, currentTime, "", true);
+    }
+
+    function executeSpamDeletion(uint256 proposalId) public {
+        require(spamDeletionProposals[proposalId].executed == false, "Already executed");
+        spamDeletionProposals[proposalId].executed = true;
+        bytes32 identifier = SpamGuardIdentifierLib._constructIdentifier(proposalId);
+
+        (bool hasPrice, int256 resolutionPrice, ) =
+            _getPriceOrError(identifier, spamDeletionProposals[proposalId].requestTime, "");
+        require(hasPrice, "Price not yet resolved");
+
+        // If the price is 1e18 then the spam deletion request was correctly voted on to delete the requests.
+        if (resolutionPrice == 1e18) {
+            // Delete the price requests associated with the spam.
+            for (uint256 i = 0; i < spamDeletionProposals[proposalId].spamRequestIndices.length; i++) {
+                uint256 startIndex = spamDeletionProposals[proposalId].spamRequestIndices[uint256(i)][0];
+                uint256 endIndex = spamDeletionProposals[proposalId].spamRequestIndices[uint256(i)][1];
+                for (uint256 j = startIndex; j <= endIndex; j++) {
+                    bytes32 requestId = priceRequestIds[j].requestId;
+                    // Remove from pendingPriceRequests.
+                    uint256 lastIndex = pendingPriceRequests.length - 1;
+                    PriceRequest storage lastPriceRequest = priceRequests[pendingPriceRequests[lastIndex]];
+                    lastPriceRequest.index = priceRequests[requestId].index;
+                    pendingPriceRequests[priceRequests[requestId].index] = pendingPriceRequests[lastIndex];
+                    pendingPriceRequests.pop();
+
+                    // Remove the request from the priceRequests mapping.
+                    delete priceRequests[requestId];
+                }
+
+                // Set the deletion request jump mapping. This enables the for loops that iterate over requests to skip
+                // the deleted requests via a "jump" over the removed elements from the array.
+                deletedRequestJumpMapping[startIndex] = endIndex;
+            }
+
+            // Return the spamDeletionProposalBond.
+            votingToken.transfer(spamDeletionProposals[proposalId].proposer, spamDeletionProposalBond);
+        }
+        // Else, the spam deletion request was voted down. In this case we send the spamDeletionProposalBond to the store.
+        else {
+            votingToken.transfer(finder.getImplementationAddress(OracleInterfaces.Store), spamDeletionProposalBond);
```

Comment:
We probably need to be careful here. Is it possible for the spamDeletionProposalBond to change between the proposal and here? Maybe it's fine to just store the initial bond value in the struct since gas isn't a concern (this would be almost never called).